src/license_expression/__init__.py

```diff
@@ -599,6 +599,35 @@ def simple_tokenizer(self, expression):
                         sym = LicenseSymbol(key=sym_or_op)
                     yield Token(start, end, sym_or_op, sym)
 
+    def dedup(self, expression):
+        """
+        Return a de-dupulicated expression
+        """
+        l = Licensing()
+        exp = l.parse(expression)
+        operator = vars(exp)['operator']
+        dedup_expression = ''
+        expression_list = []
+        for arg in vars(exp)['args']:
+            if isinstance(arg, l.AND) or isinstance(arg, l.OR):
+                # Run this recursive function if there is another AND/OR expression
+                # and add the expression to the expression_list. 
+                expression_list.append(self.dedup(arg))
+            else:
+                # Get the license key from the expression.
+                exp_key = l.license_keys(arg)[0]
+                # Add the license key to the expression_list if it's not already
+                # present.
+                if not exp_key in expression_list:
+                    expression_list.append(exp_key)
+        # The operator can only be either ' AND ' or ' OR '
+        if operator == ' AND ':
+            dedup_expression = ' AND '.join(expression_list)
```

Comment:
We likely wat to reuse /copy something like https://github.com/nexB/scancode-toolkit/blob/f3ef5b4ad823577e507d673a7fbc65d5efe4f6af/src/packagedcode/utils.py#L132 ... which should be a method here instead?

Reply:
Added a `combine_expression` function (https://github.com/nexB/license-expression/pull/53/commits/5af569ff64edefb707545c73c8f2fb9e45789217)